File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/masterdata/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.masterdata.config;

import io.zenwave360.example.clinicaltool.modules.masterdata.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.inmemory.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final MasterDataRepository masterDataRepository = new MasterDataRepositoryInMemory();

    @Bean
    @Primary
    public <T extends MasterDataRepository> T masterDataRepository() {
        return (T) masterDataRepository;
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.masterdata.config

import io.zenwave360.example.clinicaltool.modules.masterdata.*
import io.zenwave360.example.clinicaltool.modules.masterdata.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val masterDataRepository = MasterDataRepositoryInMemory()

    @Bean
    @Primary
    fun masterDataRepository(): MasterDataRepositoryInMemory {
        return masterDataRepository
    }
}
